### DISCUSSION TITLE
`pixi` for package management?

### DISCUSSION THREAD

**Discussion by jGaboardi** [pysal/libpysal#690] [General]
Saw [this blog post](https://prefix.dev/blog/pixi_a_fast_conda_alternative) by @wolfv on another emerging package management tool. I'm not at all suggesting we leap to [`pixi`](https://github.com/prefix-dev/pixi), but it may be something to further look at in the future. Mostly putting this here as a discussion point and note to self.

**Comment by knaaptime** [comment]
would probably speed up our tests https://pixi.sh/dev/advanced/github_actions/

**Reply by jGaboardi** [reply]
I'd say let's give it a try trial run while keeping our current infrastructure intact. Also, before we start *any* work on that it would be prudent to create a tagged point in time (not a release, just a tag) so we can easily revert back if needed.

**Reply by sjsrey** [reply]
I would vote for trying it out on one package and giving it a serious road test before we adopt it ecosystem-wide. 

And, maybe we should start adopting some sequencing for these changes. I get behind on the changes appearing in different packages at different times, so if we standardize it would definitely help.

**Reply by jGaboardi** [reply]
And before we take on any more infrastructure-type enhancements, we probably want to close out existing standardization issues:

* https://github.com/pysal/pysal/issues/1356
* https://github.com/pysal/pysal/issues/1341
* https://github.com/pysal/pysal/issues/1340
* https://github.com/pysal/pysal/issues/1341
* https://github.com/pysal/pysal/issues/1339
* https://github.com/pysal/pysal/issues/1338
* https://github.com/pysal/pysal/issues/1341
* https://github.com/pysal/pysal/issues/1298
* https://github.com/pysal/pysal/issues/1296
* https://github.com/pysal/pysal/issues/1277

**Reply by martinfleis** [reply]
I am not against testing it but I don't think it serves our purpose. We need to test against fresh environments (many of them). The whole point of Pixi is to create one (or possibly more) and freeze those using a lock file. I think that micromamba is still the best solution out there for this, despite the hiccups with the latest release on Windows. I am saying that as someone who uses Pixi for local development, teaching and also for some reproducible environments for research.

**Reply by knaaptime** [reply]
not dying to change anything, just that pixi says its 4x faster than conda so that could speed up the ci

**Reply by martinfleis** [reply]
That is true, compare https://github.com/martinfleis/sds/actions/runs/11271053917/job/31343106000 and https://github.com/martinfleis/sds/actions/runs/11270968877/job/31342850682. 38s with micromamba vs 10s with pixi. The action is also there for a reference.
